libtransact/src/scheduler/mod.rs

```diff
@@ -14,40 +15,124 @@
  * limitations under the License.
  * -----------------------------------------------------------------------------
  */
+
+//! Batch scheduling with transaction execution APIs
+//!
+//! A `Scheduler` is used to execute one or more `Batch` objects, receiving
+//! a `BatchExecutionResult` as a result of the execution of a `Batch`.  A `Batch` contains one or
+//! more `Transaction` objects; each `Transaction` within a `Batch` is executed and then
+//! consolidated to generate a `BatchExecutionResult`.
+//!
+//! In order for a `Scheduler` to execute batches, its associated `SchedulerExecutionInterface`
+//! must be consumed by a component responsible for iterating over the `Transaction`s and providing
+//! `TransactionExecutionResult`s back to the `Scheduler` via the `SchedulerExecutionInterface`.
+
 pub mod tree;
 
+use crate::batch::BatchPair;
 use crate::context::ContextId;
 use crate::execution::adapter::ExecutionResult;
+use crate::receipts::TransactionReceipt;
 use crate::transaction::TransactionPair;
 
-/// The `TransactionPair` along with the information needed to execute that
-/// `Transaction`
+/// A transation and associated information required to execute it.
 pub struct ExecutionTask {
     pair: TransactionPair,
     context_id: ContextId,
 }
 
 impl ExecutionTask {
+    /// Create a new `ExecutionPair`.
     pub fn new(pair: TransactionPair, context_id: ContextId) -> Self {
         ExecutionTask { pair, context_id }
     }
 
+    /// The transaction to be executed.
     pub fn pair(&self) -> &TransactionPair {
         &self.pair
     }
 
+    /// The identifier of the context to be used when accessing state.
     pub fn context_id(&self) -> &ContextId {
         &self.context_id
     }
 
+    /// Decompose into its components.
     pub fn take(self) -> (TransactionPair, ContextId) {
         (self.pair, self.context_id)
     }
 }
 
+/// Result from executing an invalid transaction.
+pub struct InvalidTransactionResult {
+    /// Transaction identifier.
+    pub transaction_id: String,
+
+    /// Human-readable reason explaining why the transaction was invalid.
+    pub error_message: String,
+
+    /// Transaction-specific error data which can be interpreted by clients
+    /// familiar with this transaction's family.
+    pub error_data: Vec<u8>,
+}
+
+/// Result from executing a transaction.
+pub enum TransactionExecutionResult {
+    /// The transation was invalid.
+    Invalid(InvalidTransactionResult),
+
+    /// The transation was valid and execution produced a TransactionReceipt.
+    Valid(TransactionReceipt<String, Vec<u8>>),
+}
+
+/// Result of executing a batch.
+pub struct BatchExecutionResult {
+    /// The `BatchPair` which was executed.
+    pub batch: BatchPair,
+
+    /// The results for each transaction in the batch.
+    pub results: Vec<TransactionExecutionResult>,
```

Comment:
This rename isn't in this commit, but in the other commit.  It seems like this should be in a third, ExecutionResult rename commit.